internal/service/credentials/resource_credential_type.go

```diff
@@ -0,0 +1,844 @@
+							// Required until P1Creds follows the standard PingOne image handling capability.
+							// Attempts of other stop-gap mechanisms to detect and update Content-Type yielded inconsistent results.
+							stringvalidator.RegexMatches(regexp.MustCompile(`^data:image\/(\w+);base64,`), "base64encoded image must indclude Content-type prefix, such as data:image/jpeg;base64, data:image/svg;base64, or data:image/png;base64."),
+							IsBase64Encoded(),
+							IsRequiredIfRegexMatchesPathValue(
+								regexp.MustCompile(`\${logoImage}`),
+								"The metadata.card_color argument is required because the ${$logoImage} element is defined in the card_design_template.",
+								path.MatchRoot("card_design_template"),
+							),
+							RegexMatchesPathValue(
+								regexp.MustCompile(`\${logoImage}`),
+								"The metadata.logo_image argument is defined but the card_design_template does not have a ${logoImage} element.",
+								path.MatchRoot("card_design_template"),
+							),
+						},
+					},
+
+					"name": schema.StringAttribute{
+						Description: "Name of the credential.",
+						Optional:    true,
+						Validators: []validator.String{
+							stringvalidator.LengthAtLeast(attrMinLength),
+							IsRequiredIfRegexMatchesPathValue(
+								regexp.MustCompile(`\${cardTitle}`),
+								"The metadata.name argument is required because the ${$cardTitle} element is defined in the card_design_template.",
+								path.MatchRoot("card_design_template"),
+							),
+							RegexMatchesPathValue(
+								regexp.MustCompile(`\${cardTitle}`),
+								"The metadata.name argument is defined but the card_design_template does not have a ${cardTitle} element.",
+								path.MatchRoot("card_design_template"),
+							),
+						},
+					},
+
+					"text_color": schema.StringAttribute{
+						Description: "A string containing a 6-digit hexadecimal color code specifying the color of the credential text.",
+						Optional:    true,
+						Validators: []validator.String{
+							stringvalidator.RegexMatches(
+								regexp.MustCompile(
+									`^#([A-Fa-f0-9]{6})$`),
+								"expected value to contain a valid 6-digit hexadecimal color code, prefixed with a hash (#) symbol."),
+							IsRequiredIfRegexMatchesPathValue(
+								regexp.MustCompile(`\${textColor}`),
+								"The metadata.text_color argument is required because the ${$textColor} element is defined in the card_design_template.",
+								path.MatchRoot("card_design_template"),
+							),
+							RegexMatchesPathValue(
+								regexp.MustCompile(`\${textColor}`),
+								"The metadata.text_color argument is defined but the card_design_template does not have a ${textColor} element.",
+								path.MatchRoot("card_design_template"),
+							),
+						},
+					},
+
+					"version": schema.Int64Attribute{
+						Description: "Number version of this credential.",
+						Required:    true, // not required in schema, but credentials will not display in P1 admin console if not provided
```

Comment:
Expect a JIRA is needed. Because of that, should this be incremented by the provider internally and not by the user?  If so, we can change this to `Computed` or `Optional`+`Computed` if we still want the user to be able to define a value

Reply:
Product is aware (do not have JIRA number off hand, but will double check).  Modified to `Computed` with a `Default` value of 5, which is the default value set by the P1 admin console.